### PR TITLE
API to check PLDM running status

### PR DESCRIPTION
**Pull request by SunnySrivastava1984**
The commit adds implementation to check if the PLDM service is running or not.

### CODE REVIEW

**Review by jinuthomas — APPROVED** [review]

**Review by branupama — CHANGES_REQUESTED** [review]

File: include/utility/dbus_utility.hpp [hunk 1]
```diff
@@ -197,5 +197,39 @@ inline bool callPIM(types::ObjectMap&& objectMap)
...
+inline bool checkServiceStatus(const std::string& i_serviceName)
+{
+    bool l_retVal = false;
+
+    try
+    {
+        auto l_bus = sdbusplus::bus::new_default();
+        auto l_method = l_bus.new_method_call(
+            "org.freedesktop.DBus", "/org/freedesktop/DBus",
+            "org.freedesktop.DBus", "NameHasOwner");
```
Comment (PriyangaRamasamy) [inline include/utility/dbus_utility.hpp:220]: why did you change from NameOwnerChanged to NameHasOwner?

File: include/utility/dbus_utility.hpp [hunk 2]
```diff
@@ -197,5 +197,39 @@ inline bool callPIM(types::ObjectMap&& objectMap)
...
 }
+
+/**
+ * @brief API to check D-Bus service status.
+ *
+ * Any failure in calling the method "NameHasOwner" implies that the service is
+ * not in a running state. Hence the API returns false in case of any exception
+ * as well.
+ *
+ * @param[in] i_serviceName - D-Bus service name whose status is to be checked.
```
[the remark this answers is not in the document]
Comment (SunnySrivastava1984) [reply include/utility/dbus_utility.hpp:208]: Is the current one wrong?

File: include/utility/dbus_utility.hpp [hunk 3]
```diff
@@ -197,5 +197,39 @@ inline bool callPIM(types::ObjectMap&& objectMap)
     }
     return true;
 }
+
+/**
+ * @brief API to check D-Bus service status.
+ *
+ * Any failure in calling the method "NameHasOwner" implies that the service is
+ * not in a running state. Hence the API returns false in case of any exception
+ * as well.
```
[the remark this answers is not in the document]
Comment (SunnySrivastava1984) [reply include/utility/dbus_utility.hpp:206]: "As well" has been mentioned here to clarify that even in case of exception it will return false.

File: include/utility/dbus_utility.hpp [hunk 4]
```diff
@@ -197,5 +197,39 @@ inline bool callPIM(types::ObjectMap&& objectMap)
     }
     return true;
 }
+
+/**
+ * @brief API to check D-Bus service status.
+ *
+ * Any failure in calling the method "NameHasOwner" implies that the service is
+ * not in a running state. Hence the API returns false in case of any exception
```
[the remark this answers is not in the document]
Comment (SunnySrivastava1984) [reply include/utility/dbus_utility.hpp:205]: Is the current one in correct?

File: include/utility/dbus_utility.hpp (hunk 1 above)
Comment (SunnySrivastava1984) [reply include/utility/dbus_utility.hpp:220]: Did I change that?

File: include/utility/dbus_utility.hpp [hunk 5]
```diff
@@ -197,5 +197,39 @@ inline bool callPIM(types::ObjectMap&& objectMap)
...
+ * as well.
+ *
+ * @param[in] i_serviceName - D-Bus service name whose status is to be checked.
+ * @return bool - True if the service is running false otherwise.
```
[the remark this answers is not in the document]
Comment (SunnySrivastava1984) [reply include/utility/dbus_utility.hpp:209]: I will fix that.

File: include/utility/dbus_utility.hpp (hunk 3 above)
[the remark this answers is not in the document]
Comment (branupama) [reply include/utility/dbus_utility.hpp:206]: okay.

File: include/utility/dbus_utility.hpp (hunk 2 above)
[the remark this answers is not in the document]
Comment (branupama) [reply include/utility/dbus_utility.hpp:208]: the current is not wrong, but felt 'has' is better here.

File: include/utility/dbus_utility.hpp (hunk 4 above)
[the remark this answers is not in the document]
Comment (branupama) [reply include/utility/dbus_utility.hpp:205]: its not wrong.

**Review by branupama — APPROVED** [review]

File: include/utility/dbus_utility.hpp [hunk 6]
```diff
@@ -197,5 +197,39 @@ inline bool callPIM(types::ObjectMap&& objectMap)
...
+ */
+inline bool checkServiceStatus(const std::string& i_serviceName)
```
Comment (PriyangaRamasamy) [inline include/utility/dbus_utility.hpp:211]: isServiceRunning() could be the better name for this method, as it returns bool.

Comment (SunnySrivastava1984) [reply include/utility/dbus_utility.hpp]: Done.

File: include/biosHandler.hpp [hunk 7]
```diff
@@ -29,8 +29,9 @@ class BiosHandler
      *
      * @param[in] connection - Asio connection object.
      */
-    BiosHandler(std::shared_ptr<sdbusplus::asio::connection>& connection) :
-        m_asioConn(connection)
+    BiosHandler(
+        const std::shared_ptr<sdbusplus::asio::connection>& i_connection) :
```
Comment (PriyangaRamasamy) [inline include/biosHandler.hpp:33]: if you do a rebase, why it is showing as green for line 33 and 51.
